hadoop-ozone/ozone-manager/src/test/java/org/apache/hadoop/ozone/om/TestKeyManagerUnit.java

```diff
@@ -0,0 +1,93 @@
+package org.apache.hadoop.ozone.om;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import org.apache.hadoop.hdds.HddsConfigKeys;
+import org.apache.hadoop.hdds.conf.OzoneConfiguration;
+import org.apache.hadoop.hdds.protocol.StorageType;
+import org.apache.hadoop.hdds.protocol.proto.HddsProtos.ReplicationFactor;
+import org.apache.hadoop.hdds.protocol.proto.HddsProtos.ReplicationType;
+import org.apache.hadoop.hdds.scm.protocol.ScmBlockLocationProtocol;
+import org.apache.hadoop.ozone.om.helpers.OmBucketInfo;
+import org.apache.hadoop.ozone.om.helpers.OmKeyArgs;
+import org.apache.hadoop.ozone.om.helpers.OmKeyArgs.Builder;
+import org.apache.hadoop.ozone.om.helpers.OmMultipartInfo;
+import org.apache.hadoop.ozone.om.helpers.OmMultipartUploadListParts;
+import org.apache.hadoop.ozone.security.OzoneBlockTokenSecretManager;
+import org.apache.hadoop.test.GenericTestUtils;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+/**
+ * Unit test key manager.
+ */
+public class TestKeyManagerUnit {
```

Comment:
We already have a class for KeyManagerImpl unit tests "TestKeyManagerImpl". I think we can use that or make this one specific for the s3 apis?

Reply:
KeyManagerImpl is more like an integration test (and part of the integration-test project). It starts an almost real SCM with all the sub-components and network components which takes more time and have more risk to be flaky.

I prefer to use unit tests when possible with limited number of components, which is more lightweight and faster.

This new unit test uses only on real dependency, the metadata manager. While it's still realistic, it's faster and safer.

To avoid the name collision (same unit test in two different projects) I added the Unit postfix. 

Long-term I would suggest to migrate the existing 'integration test' of the KeyManager class to this class (if possible)

Reply:
Yes, that makes sense. We can move some unit tests which do not require scm and other components involvement later into this test.